src/Form.php

```diff
@@ -0,0 +1,378 @@
+     * declare a label `First Name` which is more user-friendly and can be displayed to end users.
+     *
+     * By default an attribute label is generated using {@see generateAttributeLabel()}. This method allows you to
+     * explicitly specify attribute labels.
+     *
+     * Note, in order to inherit labels defined in the parent class, a child class needs to merge the parent labels
+     * with child labels using functions such as `array_merge()`.
+     *
+     * @return array attribute labels (name => label)
+     *
+     * {@see generateAttributeLabel()}
+     */
+    public function attributesLabels(): array
+    {
+        return [];
+    }
+
+    /**
+     * Removes errors for all attributes or a single attribute.
+     *
+     * @param string|null $attribute attribute name. Use null to remove errors for all attributes.
+     */
+    public function clearErrors(?string $attribute = null): void
+    {
+        if ($attribute === null) {
+            $this->errors = [];
+        } else {
+            unset($this->errors[$attribute]);
+        }
+    }
+
+    /**
+     * Returns attribute values.
+     *
+     * @return array attribute values (name => value).
+     */
+    public function getAttributes(): array
+    {
+        return $this->attributes;
+    }
+
+    /**
+     * Returns the text hint for the specified attribute.
+     *
+     * @param string $attribute the attribute name.
+     *
+     * @return string the attribute hint.
+     *
+     * {@see getAttributesHints()}
+     */
+    public function getAttributeHint(string $attribute): string
+    {
+        $hints = $this->getAttributesHints();
+
+        return isset($hints[$attribute]) ? $hints[$attribute] : '';
+    }
+
+    /**
+     * Returns the attribute hints.
+     *
+     * Attribute hints are mainly used for display purpose. For example, given an attribute `isPublic`, we can declare
+     * a hint `Whether the post should be visible for not logged in users`, which provides user-friendly description of
+     * the attribute meaning and can be displayed to end users.
+     *
+     * Unlike label hint will not be generated, if its explicit declaration is omitted.
+     *
+     * Note, in order to inherit hints defined in the parent class, a child class needs to merge the parent hints with
+     * child hints using functions such as `array_merge()`.
+     *
+     * @return array attribute hints (name => hint)
+     */
+    public function getAttributesHints(): array
+    {
+        return [];
+    }
+
+    /**
+     * Returns the text label for the specified attribute.
+     *
+     * @param string $attribute the attribute name.
+     *
+     * @return string the attribute label.
+     *
+     * {@see generateAttributeLabel()}
+     * {@see attributesLabels()}
+     */
+    public function getAttributeLabel(string $attribute): string
+    {
+        $this->attributesLabels = $this->getAttributesLabels();
```

Comment:
Why you need to update a local property?

Reply:
done.